storybookjs/storybook
# addon-knobs disableQueryParams

Issue: #9027 

## What I did

I added a decorator option that lets you disable the ability to pre-populate addon-knob values with query strings. 

## How to test

Build storybook locally and link it to an active project using `yarn`. Once linked add the option to your configuration file.

```javascript
defaultView.story = {
  parameters: {
    knobs: {
      disableQueryParams: true,
    }
  }
};
```

Navigate to a story with a knob. If the story has a knob named `label` then add `?knob-label=hello` to the end of the URL. It should remain as the default value and not take the query string. If the `disableQueryParams` option is not `true` then it _will_ pre-populate the values.

This is useful for certain stories that let you modify the href value of an anchor tag. You can essentially create a pattern like so `?knob-url=javascript alert('hello')` to execute a XSS attack depending on where the Storybook instance is hosted. 

- Is this testable with Jest or Chromatic screenshots? **No**
- Does this need a new example in the kitchen sink apps? **No**
- Does this need an update to the documentation? **Yes**
